internal/command/jsonplan/action_invocations.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package jsonplan
+
+import (
+	"github.com/hashicorp/terraform/internal/plans"
+)
+
+type ActionInvocation struct {
+	// Address is the absolute action address
```

Comment:
I think we will need the address of the triggering resource, the trigger block index and the index for the event here to clearly link the action invocation back to its place in the config and to be able to base apply on it. I think we could get if it's run before or after from this info as well by looking at the config so we don't necessarily need it in the plan. 

That being said, we can either add these now or in a follow up PR that integrates this with the plan action changes. The heavy lifting is done so I'm also ok with folding this into the upcoming work 👍

Reply:
Yep, agreed  - I didn't do it here because I'd have to rework the hooks too, and also I didn't have a good picture of when or where we would build that info